osu.Game/Overlays/ChatOverlay.cs

```diff
@@ -394,6 +394,15 @@ private void addChannel(Channel channel)
             {
                 careChannels.Add(channel);
                 channelTabs.AddItem(channel);
+
+                if (channel.Type == ChannelType.Public && !channel.Joined)
+                {
+                    var req = new JoinChannelRequest(channel, api.LocalUser);
+                    req.Success += () => addChannel(channel);
+                    req.Failure += ex => removeChannel(channel);
```

Comment:
This will fail in 100% of cases (tabcontrol throws if the item doesn't exist in the collection).

Reply:
as discussed, should be ok.